lib/adapters/mongodb/index.js

```diff
@@ -0,0 +1,138 @@
+'use strict'
+
+const Hapi = require('hapi')
+const Boom = require('boom')
+const _ = require('lodash')
+const mongoose = require('mongoose')
+const converters = require('./converters')()
+const utils = require('./utils')()
+
+mongoose.Promise = require('bluebird')
+
+module.exports = function (options) {
+
+    const models = {}
+
+    const connect = function(cb) {
+        mongoose.connect(options.mongodbUrl, cb)
+    }
+
+    const disconnect = function(cb) {
+        //clear out events
+        mongoose.connection._events = {}
+        mongoose.disconnect(cb)
+    }
+    
+    mongoose.connection.on('error', connect)
+
+    const find = function (type, req) {
+
+        const model = models[type]
+        const query = req.query
+        const limit = (query.page && query.page.limit) || 1000
+        const skip = (query.page && query.page.offset) || 0
+        const sort = converters.toMongooseSort(query.sort)
+        const include = query.include && query.include.split(',')
+        var predicate = converters.toMongoosePredicate(query)
+        return model.find(predicate).skip(skip).sort(sort).limit(limit).lean().exec()
+            .then((resources)=> {
+                let data  = converters.toJsonApi(resources);
+                if (include) {
+                    data = _.map(data, (datum) => {
```

Comment:
See comment below about mixup of jsonapi vocabulary. This is fields behaviour, not includes